### ISSUE TITLE
[dialog] ?attr/colorControlHighlight used for android:background

### ISSUE THREAD

**Issue by msimonides** [material-components/material-components-android#2455]
**Description:** `mtrl_alert_dialog.xml` and `m3_alert_dialog.xml` use `?attr/colorControlHighlight` for `android:background`

`?attr/colorControlHighlight` is a color value while `android:background` expects a drawable. Using simple color values for `android:background` works but using a color selector causes a crash because it's being interpreted as a drawable.

This makes it impossible to set color ripples to anything else than primitive color values.
A common thing is to use a color selector to apply an alpha to another color attribute as explained e.g. here: https://material.io/blog/android-material-theme-color (section "Color with alpha").

So basically following the guidelines from the Material Design blog with Material Components for Android leads to a crash when showing dialogs 👍

**Expected behavior:** No crash

**Source code:**
1. Create a color selector, e.g.

```
<!-- In res/color/primary_60.xml -->
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <item android:alpha="0.6" android:color="?attr/colorPrimary" />
</selector>
```

2. In your app theme set `colorControlHighlight`:

```
<item name="colorControlHighlight">@color/primary_60.xml</item>
```

3. Create and show a Material dialog:

```
MaterialAlertDialogBuilder(context)
    ...
    .show()
```

4. Crash

**Android API version:** ~~30 - possibly all~~ 26 and 28, possibly 28 and lower. 29+ works fine.

**Material Library version:** master, the issue can be clearly seen in source code. Search for `colorControlHighlight` in `mrtl_alert_dialog.xml` and `m3_alert_dialog.xml` files in this repository.

**Device:** any? reproducible on an emulator.

**Suggested fixes:**
- define a proper color drawable that uses `?attr/colorControlHighlight` and set that in `android:background`,
- use a different color - why would a ripple color be used for backgrounds?

**Comment by drchen** [comment]
Did you actually test it and it crashed? I cannot reproduce the crash actually. I kind of think AppCompat handles it?

**Comment by msimonides** [comment]
I'm sorry, I mixed up the API versions I tested :(
I get the crash on API26 and API28. There is no crash on API29 and API30.

I also created a simple project (attached) to rule out the possibility that this is something in the more complex app.

[MaterialDialog.zip](https://github.com/material-components/material-components-android/files/7468763/MaterialDialog.zip)

Here's the stack trace I get:
```
2021-11-03 16:29:28.426 6160-6160/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.materialdialog, PID: 6160
    android.view.InflateException: Binary XML file line #38: Binary XML file line #38: Error inflating class android.view.View
    Caused by: android.view.InflateException: Binary XML file line #38: Error inflating class android.view.View
    Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:334)
        at android.view.LayoutInflater.createView(LayoutInflater.java:647)
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:703)
        at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:68)
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:720)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:788)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:699)
        at androidx.appcompat.app.AppCompatDialog.setContentView(AppCompatDialog.java:95)
        at androidx.appcompat.app.AlertController.installContent(AlertController.java:232)
        at androidx.appcompat.app.AlertDialog.onCreate(AlertDialog.java:279)
        at android.app.Dialog.dispatchOnCreate(Dialog.java:403)
        at android.app.Dialog.show(Dialog.java:302)
        at androidx.appcompat.app.AlertDialog$Builder.show(AlertDialog.java:1009)
        at com.example.materialdialog.MainActivity.onCreate$lambda-0(MainActivity.kt:16)
        at com.example.materialdialog.MainActivity.$r8$lambda$n7I7NgMmzgnN4btXDHkcLd85pbA(Unknown Source:0)
        at com.example.materialdialog.MainActivity$$ExternalSyntheticLambda0.onClick(Unknown Source:2)
        at android.view.View.performClick(View.java:6256)
        at com.google.android.material.button.MaterialButton.performClick(MaterialButton.java:1119)
        at android.view.View$PerformClick.run(View.java:24701)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
     Caused by: android.content.res.Resources$NotFoundException: Drawable com.example.materialdialog:color/purple_500_alpha with resource ID #0x7f0500c0
     Caused by: android.content.res.Resources$NotFoundException: File res/color/purple_500_alpha.xml from drawable resource ID #0x7f0500c0
        at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:787)
        at android.content.res.ResourcesImpl.loadDrawable(ResourcesImpl.java:609)
        at android.content.res.Resources.loadDrawable(Resources.java:885)
        at android.content.res.TypedArray.getDrawableForDensity(TypedArray.java:953)
        at android.content.res.TypedArray.getDrawable(TypedArray.java:928)
        at android.view.View.<init>(View.java:4730)
        at android.view.View.<init>(View.java:4631)
        at android.view.View.<init>(View.java:4610)
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:334)
        at android.view.LayoutInflater.createView(LayoutInflater.java:647)
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:703)
        at com.android.internal.policy.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:68)
2021-11-03 16:29:28.428 6160-6160/? E/AndroidRuntime:     at android.view.LayoutInflater.onCreateView(LayoutInflater.java:720)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:788)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:699)
        at androidx.appcompat.app.AppCompatDialog.setContentView(AppCompatDialog.java:95)
        at androidx.appcompat.app.AlertController.installContent(AlertController.java:232)
        at androidx.appcompat.app.AlertDialog.onCreate(AlertDialog.java:279)
        at android.app.Dialog.dispatchOnCreate(Dialog.java:403)
        at android.app.Dialog.show(Dialog.java:302)
        at androidx.appcompat.app.AlertDialog$Builder.show(AlertDialog.java:1009)
        at com.example.materialdialog.MainActivity.onCreate$lambda-0(MainActivity.kt:16)
        at com.example.materialdialog.MainActivity.$r8$lambda$n7I7NgMmzgnN4btXDHkcLd85pbA(Unknown Source:0)
        at com.example.materialdialog.MainActivity$$ExternalSyntheticLambda0.onClick(Unknown Source:2)
        at android.view.View.performClick(View.java:6256)
        at com.google.android.material.button.MaterialButton.performClick(MaterialButton.java:1119)
        at android.view.View$PerformClick.run(View.java:24701)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
     Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #3: <item> tag requires a 'drawable' attribute or child tag defining a drawable
        at android.graphics.drawable.StateListDrawable.inflateChildElements(StateListDrawable.java:189)
        at android.graphics.drawable.StateListDrawable.inflate(StateListDrawable.java:122)
        at android.graphics.drawable.DrawableInflater.inflateFromXmlForDensity(DrawableInflater.java:142)
        at android.graphics.drawable.Drawable.createFromXmlInnerForDensity(Drawable.java:1295)
        at android.graphics.drawable.Drawable.createFromXmlForDensity(Drawable.java:1254)
        at android.content.res.ResourcesImpl.loadDrawableForCookie(ResourcesImpl.java:774)
        	... 42 more
```

**Comment by drchen** [comment]
Thanks for the clarification! That makes more sense. : )

**Comment by SergeyKudryashov** [comment]
Isn't fixed in the version [1.6.0-rc01](https://github.com/material-components/material-components-android/releases/tag/1.6.0-rc01).
